### DISCUSSION TITLE
How quickly can you access the thumbnail for a smart event?

### DISCUSSION THREAD

**Discussion by B-Hartley** [briis/unifiprotect#393] [General]
I’m on beta 2.
I’m using Telegram Bot to send a thumbnail from a smart person or vehicle detection to my phone.
However it often seems to fail and from experimentation I think it is because the thumbnail is somehow “not ready”.
If I have a delay before I send the thumbnail it seems to be more successful.

Does this make sense?

What does the API do, if I request a thumbnail that hasn’t been stored yet, or is that even possible.

if it’s not that, do you have any thoughts on why I’m often not receiving the photo.

Will switch to the new detected object sensor soon and see how that goes.

**Comment by AngellusMortis** [comment]
I was just doing some testing of this as well this morning. The event thumbnails are not generated until the event _ends_. So if you want something real time, you will need to keep using the entity picture from the camera entity.

Otherwise, you can wait until the sensor turns off and pull the value from the old_state.


It is really frustrating, but the thumbs are proxies straight to UniFi Protect, so it is the best they are giving us.
